Controllers/PhoneBookController.cs

```diff
@@ -38,22 +32,21 @@ public async Task<IActionResult> Index(string currentFilter, string searchString
 
             ViewBag.CurrentFilter = searchString;
 
-            var entries = from e in _context.Entries
-                select e;
-                    
-                    
+            var entries = await _context.Entries.Select(e => e).ToListAsync();
```

Comment:
it seems to be not reasonable to materialize (enumerate to list)  the whole DbSet since we got pagination and search string

I would propose to filter by the search string, then to paginate and only then enumerate to list

Reply:
It seems like this is the most optimal solution: [commit](https://github.com/rumkit/PhoneEdit/pull/8/commits/0eadff2a1636fdd155b69ea36e9e393b2e339425), but to implement this method, I had to adjust the connection string, as the 'StringComparison' parameter is not supported by default. Now the connection string looks like this: 

```cs
options.UseMySql(phoneBookContextSecrets,ServerVersion.AutoDetect(phoneBookContextSecrets), 
         b => b.EnableStringComparisonTranslations()));
```

Did I solve the task correctly?

Reply:
actually just Contains(string) should do the trick https://stackoverflow.com/questions/2369022/using-contains-in-linq-to-sql
the overload taking StringComparison enum as the 2nd parameter turns out not to be compatible with SLQ itself

Reply:
As it turned out, this trick wasn't necessary and everything works without it. Such a trick was required before the previous code change. I did as follows:

PhoneBookController.cs
```c#
if (!string.IsNullOrEmpty(searchString))
{
    entriesQuery = entriesQuery.Where(e =>
        e.Name.Contains(searchString) ||
        e.Mail.Contains(searchString) ||
        e.Room.Contains(searchString) ||
        e.LocalPhoneNumber.Contains(searchString) ||
        e.PersonnelNumber.Contains(searchString));
}
```

Program.cs
```c#
builder.Services.AddDbContext<PhonebookContext>(options => 
    options.UseMySql(phoneBookContextSecrets,ServerVersion.AutoDetect(phoneBookContextSecrets)));
```